Automattic/wp-calypso
# Gutenberg: Simplify new connection addition, cleanup

This PR suggests removing the list of services to connect in favor of a single link when there are no connections yet, as suggested in pafL3P-7k-p2 #comment-489

This also results in removing some boilerplate and data store code, as it's no longer used.

#### Changes proposed in this Pull Request

* Add support for providing additional classes to `<PublicizeSettingsButton />`.
* Use `<PublicizeSettingsButton />` instead of `<PublicizeNoConnections />`.
* Remove unused `<PublicizeNoConnections />`.
* Remove any services handling from the data store as it's no longer used.
* Flatten state, as it now contains only connections.
* Update copy of the new connection link. 

#### Screenshots

Before (no connections):
![](https://cldup.com/gy7G-xlxFy.png)

After (no connections:
![](https://cldup.com/iCzXiXA0c3.png)

Before (with connections):
![](https://cldup.com/9cco8m71EK.png)

After (with connections):
![](https://cldup.com/3F_IryPzKi.png)

#### Testing instructions

* Spin up a JN site with this branch and Jetpack's `add/publicize-rest-api` from [this link](https://jurassic.ninja/create/?shortlived&gutenpack&gutenberg&jetpack-beta&branch=add/publicize-rest-api&calypsobranch=update/gutenberg-publicize-simplify-new-connection).
* Connect the site to WP.com.
* Start a new post, write a title and some content.
* Attempt to publish.
* Verify you can see Publicize in the pre-publish sidebar.
* Click the "Connect new service" link.
* Connect a social account.
* Close the new tab that was opened.
* Verify connections are refreshed, and you can see your connection in the list.
* Refresh the page.
* Attempt to publish.
* Verify connections load properly.
* Verify removed services code is no longer used anywhere.

Note: we're removing the **Refresh connections** link separately in #28382.
